src/availability.jl

```diff
@@ -16,25 +19,26 @@ The currently supported values for `platform` are:
 - `:darwin`: for Darwin kernel availability
 """
 struct PlatformAvailability{P}
+    appl_plat::Bool # Is this Availability object applicable on this system
     introduced::Union{Nothing, VersionNumber}
     deprecated::Union{Nothing, VersionNumber}
     obsoleted::Union{Nothing, VersionNumber}
     unavailable::Bool
```

Comment:
```suggestion
    introduced::Union{Nothing, VersionNumber}
    deprecated::Union{Nothing, VersionNumber}
    obsoleted::Union{Nothing, VersionNumber}
    applicable::Bool
    unavailable::Bool
```

- what's the purpose of the `unavailability` bool -- shouldn't that always be determined based on the introduced/deprecated/obsoleted?
- maybe it's cleaner to simply have `macos(...)` return `false` on irrelevant platforms? or to fold the `Sys.isapple` check into `is_available(::PlatformAvailability{:macOS})` and always return `false` from there?

Reply:
Yeah I don't know why I didn't just have it be checked in `is_available`, I've changed that and got rid of the new field completely.

`unavailable` is part of the Clang `PlatformAvailability`. I don't think it's very relevant here as I believed it's used for APIs shared between platforms where one interface is never available on that platform, but I added it in because it's something that Clang.jl could emit at some point.

Reply:
TL:DR: Leave `unavailable` in for consistency?

For example, looking at the Metal headers, `synchronizeResource` is marked as unavailable in iOS, and it seems like a lot of `init` functions are explicitly marked unavailable when something else should be used.

My Clang PR currently ignores things marked `unavailable` because I figured there's no point in increasing loading and precompile time for unavailable objects/properties, but that may change at some point.